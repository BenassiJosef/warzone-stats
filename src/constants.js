// Style common configuration.
export const PRIMARY_COLOUR = "#0036a8";
export const PALE_GREY = "#eef2f9";
export const FONT_FAMILY = "Poppins";
export const FONT_FAMILY_REGULAR = "Poppins-Regular";
export const TEXT_FONT_SIZE = "18px";

// console.log(process.env)

// API config settings.
export const WARZONE_API = process.env.REACT_APP_WARZONE_BASE_URL;
