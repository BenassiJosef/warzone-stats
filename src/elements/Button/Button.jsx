import styled from "styled-components";
import { PRIMARY_COLOUR, FONT_FAMILY } from "../../constants";

const Button = styled.button.attrs((props) => ({
  "data-test": `${props.intent}-button`,
}))`
  /* Adapt the colors based on primary prop */
  background: ${(props) => {
    switch (props.intent) {
      case "primary":
        return PRIMARY_COLOUR;
      case "secondary":
        return "white";
      case "link":
        return "none";
      default:
        return PRIMARY_COLOUR;
    }
  }};

  color: ${(props) => {
    switch (props.intent) {
      case "primary":
        return "white";
      case "secondary":
        return PRIMARY_COLOUR;
      case "link":
        return PRIMARY_COLOUR;
      default:
        return "white";
    }
  }};

  border-color: ${(props) => {
    switch (props.intent) {
      case "primary":
      case "secondary":
        return PRIMARY_COLOUR;
      case "link":
        return "white";
      default:
        return "white";
    }
  }};

  padding: ${(props) => {
    switch (props.intent) {
      case "primary":
        return "9px 20px 8px 20px";
      case "secondary":
        return "9px 20px 8px 20px";
      case "link":
        return 0;
      default:
        return "9px 20px 8px 20px";
    }
  }};

  border: ${(props) => {
    switch (props.intent) {
      case "primary":
        return `solid 1px ${PRIMARY_COLOUR}`;
      case "secondary":
        return `solid 1px ${PRIMARY_COLOUR}`;
      case "link":
        return "none";
      default:
        return "inherit";
    }
  }};

  text-align: ${(props) => {
    switch (props.intent) {
      case "primary":
        return "center";
      case "secondary":
        return "center";
      case "link":
        return "left";
      default:
        return "inherit";
    }
  }};

  :focus {
    outline: 0;
  }
  opacity: ${(props) => (props.disabled === true ? 0.6 : 1.0)};

  min-width: 156px;
  font-family: ${FONT_FAMILY};
  font-size: 16px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: normal;
  letter-spacing: normal;
  border-radius: 3px;
  margin-right: 16px;

  :hover {
    cursor: ${(props) => (props.disabled === true ? "not-allowed" : "pointer")};
  }
`;

export default Button;
