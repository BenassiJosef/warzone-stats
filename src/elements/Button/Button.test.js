import React from "react";
import Enzyme, { shallow } from "enzyme";
import EnzymeAdapter from "enzyme-adapter-react-16";
import "jest-styled-components";
import renderer from "react-test-renderer";
import Button from "./Button";
import { PRIMARY_COLOUR } from "../../constants";

Enzyme.configure({ adapter: new EnzymeAdapter() });

/**
 * Return ShallowWrapper conataining node(s) with the given data-test value.
 * @param {shallow} wrapper - Enzyme shallow wrapper to search within.
 * @param {string} val - Value of data-test attribute for search.
 * @returns {shallow}
 */
const findByTestAttribute = (wrapper, val) => {
  return wrapper.find(`[data-test='${val}']`);
};

test("button component renders without crashing", () => {
  const wrapper = shallow(<Button>Test me</Button>);
  expect(wrapper).toBeTruthy();
});

test("button component renders as primary by default", () => {
  const tree = renderer.create(<Button />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule("background", PRIMARY_COLOUR);
  expect(tree).toHaveStyleRule("color", "white");
  expect(tree).toHaveStyleRule("border-color", "white");
});

test("button component renders as primary by props attribute", () => {
  const wrapper = shallow(<Button intent="primary">Test me</Button>);
  // console.log(wrapper.debug());
  const testButton = findByTestAttribute(wrapper, "primary-button");

  // Check the we only have one button rendered.
  expect(testButton.length).toBe(1);

  // Check that the button actually renders the expected children.
  expect(testButton.props().children).toBe("Test me");

  // Check that the button is actually a primary button.
  expect(testButton).toHaveStyleRule("background", PRIMARY_COLOUR);
});

test("button component renders as secondary by props attribute", () => {
  const wrapper = shallow(<Button intent="secondary">Test me</Button>);
  // console.log(wrapper.debug());
  const testButton = findByTestAttribute(wrapper, "secondary-button");

  // Check the we only have one button rendered.
  expect(testButton.length).toBe(1);

  // Check that the button actually renders the expected children.
  expect(testButton.props().children).toBe("Test me");

  // Check that the button is actually a primary button.
  expect(testButton).toHaveStyleRule("background", "white");
});
