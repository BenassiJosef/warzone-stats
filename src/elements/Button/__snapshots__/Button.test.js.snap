// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button component renders as primary by default 1`] = `
.c0 {
  background: #0036a8;
  color: white;
  border-color: white;
  padding: 9px 20px 8px 20px;
  border: inherit;
  text-align: inherit;
  opacity: 1;
  min-width: 156px;
  font-family: Poppins;
  font-size: 16px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  border-radius: 3px;
  margin-right: 16px;
}

.c0:focus {
  outline: 0;
}

.c0:hover {
  cursor: pointer;
}

<button
  className="c0"
  data-test="undefined-button"
/>
`;
